test/TestServerProjectsLowLevel/dpg-customization/Customization/DPGClient.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            RequestContext requestContext = new RequestContext();
+            requestContext.CancellationToken = cancellationToken;
+
+            Response response = GetModel(mode, requestContext);
+            return Response.FromValue((Product)response, response);
+        }
+
+        /// <summary> Post either raw response as a model and pass in &apos;raw&apos; for mode, or grow up your operation to take a model instead, and put in &apos;model&apos; as mode. </summary>
+        /// <param name="mode"> The mode with which you&apos;ll be handling your returned body. &apos;raw&apos; for just dealing with the raw body, and &apos;model&apos; if you are going to convert the raw body to a customized body before returning to users. </param>
+        /// <param name="input"> Please put {&apos;hello&apos;: &apos;world!&apos;}. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> or <paramref name="input"/> is null. </exception>
+        public virtual async Task<Response<Product>> PostModelAsync(string mode, Input input, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(input, nameof(input));
+
+            // TO-DO: how to convert from model Input to requestContent
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream);
+            (input as IUtf8JsonSerializable).Write(writer);
+            writer.Flush();
+
+            RequestContext requestContext = new RequestContext();
+            requestContext.CancellationToken = cancellationToken;
+
+            Response response = await PostModelAsync("model", RequestContent.Create(stream.ToArray()), requestContext);
+            return Response.FromValue((Product)response, response);
+        }
+
+        /// <summary> Post either raw response as a model and pass in &apos;raw&apos; for mode, or grow up your operation to take a model instead, and put in &apos;model&apos; as mode. </summary>
+        /// <param name="mode"> The mode with which you&apos;ll be handling your returned body. &apos;raw&apos; for just dealing with the raw body, and &apos;model&apos; if you are going to convert the raw body to a customized body before returning to users. </param>
+        /// <param name="input"> Please put {&apos;hello&apos;: &apos;world!&apos;}. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="mode"/> or <paramref name="input"/> is null. </exception>
+        public virtual Response<Product> PostModel(string mode, Input input, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(input, nameof(input));
+
+            // TO-DO: how to convert from model Input to requestContent
```

Comment:
I will get you an answer on this.

Reply:
You can use `RequestContent.Create(input);`

Reply:
Do we have a specification of sending json? 
If we take this Input model as an example, it has small case (maybe camelcase) "[hello](https://github.com/Azure/autorest.testserver/blob/f6938f4d4d639853f03206e9c2d8bf94abcd7689/swagger/dpg-customization.json#L141)" property. In HLC generated Input.Serialization, it does have small case "hello".
But if you change the code to `RequestContent.Create(input);` , it uses default serializer which will generate uppercase "Hello".
We have to do this which is not convenient.
```
var options = new JsonSerializerOptions
{
    PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
};
RequestContent.Create(input, new JsonObjectSerializer(options);
```

Reply:
@annelo-msft and @pshao25 how about using [this](https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeedDetail.Serialization.cs#L487-L498) approach for changing Input Model -> RequestContent and Response -> Output Model?

Reply:
@ShivangiReja, yes, this is a better solution.  Thank you!

Reply:
@ShivangiReja @annelo-msft Yes, the Response -> Output Model is the same as what we call in cast. But the Input Model -> RequestContent is internal, we could use in this sample, but customer cannot use. Is this OK?

Reply:
I think it is ok to have both be internal at this point.  What is important right now is that we prove the #4 grow-up story, and providing the method to take the input model and return the output model is sufficient for that.  Thanks, @pshao25!  Let me know if you need further clarification here.